test_scripts/Polices/External_UCS_Informing_HMI/019_ATF_N_GetListofPermissions_empty_due_to_invalid_param_entityType_OnAppPermissionChange.lua

```diff
@@ -0,0 +1,142 @@
+
+--[[Local Variables]]
+local params_invalid_data =
+{
+  {param_value = "invalidValue", comment = "String"}
+  {param_value = 1.32 comment = "Float"},
+  {param_value = {}, comment = "Empty table" },
+  {param_value = { entityType = 1, entityID = 1 }, comment = "Non-empty table"},
+  {param_value = -1, comment = "OutOfLowerBound"},
+  {param_value = 129, comment = "OutOfUpperBound" },
+  {param_value = "", comment = "Empty" },
+  {param_value = nil, desc = "Null" }
+}
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_trigger_getting_device_consent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+for i = 1, #params_invalid_data do
+  Test["Precondition_PTU_and_OnAppPermissionConsent_Invalid_"..params_invalid_data[i]] = function(self)
+    local ptu_file_path = "files/jsons/Policies/Related_HMI_API/"
+    local ptu_file = "OnAppPermissionConsent_ptu.json"
+    
+    testCasesForPolicyTable:flow_SUCCEESS_EXTERNAL_PROPRIETARY(self, nil, nil, nil, ptu_file_path, nil, ptu_file)
+    
+    EXPECT_HMINOTIFICATION("SDL.OnAppPermissionChanged",{ appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+    :Do(function(_,data)
+      if (data.params.appPermissionsConsentNeeded== true) then
+        local RequestIdListOfPermissions = self.hmiConnection:SendRequest("SDL.GetListOfPermissions", {appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+          EXPECT_HMIRESPONSE(RequestIdListOfPermissions,{result = {code = 0, method = "SDL.GetListOfPermissions",
+            -- allowed: If ommited - no information about User Consent is yet found for app.
+            allowedFunctions = {
+              { name = "Location", id = 156072572},
+              { name = "Notifications", id = 1809526495}
+            },
+            externalConsentStatus = {}
+          }
+        })
+        :Do(function()
+          local ReqIDGetUserFriendlyMessage = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage",
+          {language = "EN-US", messageCodes = {"AppPermissions"}})
+          
+          EXPECT_HMIRESPONSE(ReqIDGetUserFriendlyMessage,
+          {result = {code = 0, messages = {{messageCode = "AppPermissions"}}, method = "SDL.GetUserFriendlyMessage"}})
+          :Do(function(_,_)
+            self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent",
+            {
+              appID = self.applications[config.application1.registerAppInterfaceParams.appName],
+              consentedFunctions = {
```

Comment:
@vvvakulenko, based on comment on line 80, you should update:
consentedFunctions = consentedgroups